File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/data/source/CityDataSource.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.data.source;

import android.support.annotation.NonNull;

import com.example.tsvetelinastoyanova.weatherreportrevisited.data.CityEntity;
import com.example.tsvetelinastoyanova.weatherreportrevisited.model.WeatherObject;

public interface CityDataSource {

    interface GetCityCallback {

        void onCityLoaded(CityEntity cityEntity);

        void onCityDoesNotExist();
    }

    void getCity(@NonNull String cityName, @NonNull GetCityCallback callback);

    interface AddCityCallback {

        void onCityAddedSuccessfully(CityEntity cityEntity);

        void onFail();
    }

    interface GetWeatherObjectCallback {

        void onWeatherObjectLoaded(WeatherObject weatherObject);

        void onFail();
    }

}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/data/source/CityDataSource.kt
package com.example.tsvetelinastoyanova.weatherapp.data.source

import com.example.tsvetelinastoyanova.weatherapp.data.CityEntity
import io.reactivex.Single

interface CityDataSource {

    interface GetCityCallback {

        fun onCityLoaded(cityEntity: CityEntity)

        fun onCityDoesNotExist()
    }

    fun getCity(cityName: String): Single<CityEntity>
}